avocado/utils/cpu.py

```diff
@@ -374,3 +376,49 @@ def get_pid_cpus(pid):
         except IOError:
             continue
     return list(cpus)
+
+
+def _is_smt():
+
+    """
+    Return True/False Based on if system has hyperthreading enabled
+    :return: True/False
+    :rtype: boolean
+    """
+
+    total_logical_cpus = 0
+    total_physical_cpus = 0
+    total_cores = 0
+    logical_cpus = {}
+    physical_cpus = {}
+    cores = {}
+    if get_cpu_vendor_name().startswith('power'):
+        try:
+            if b'is not SMT capable' in process.system_output("ppc64_cpu --smt"):
+                return False
+            return True
+        except process.CmdError as err:
+            logging.error("Unable to get hyperthreading info  %s", err)
```

Comment:
this can be reworded as ` Unable to get Simultaneous multithreading status` and still you need to return either `True` or `False`, for the benefit of doubt, `False` is better.

Reply:
ah yes i can reworded , but returning `False` will mislead the user  in here  command failed to run

Reply:
if then, you need to error out the test, as test/other method waiting for the return value to proceed would anyway get mislead...

Reply:
ah may be i can return `False` only as either case it not able to fetch value for SMT